libraries/textcomposer/impl/src/main/kotlin/io/element/android/libraries/textcomposer/components/RecordButton.kt

```diff
@@ -68,26 +87,36 @@ internal fun RecordButton(
         onTap = {
             onTap()
             performHapticFeedback()
+            coroutineScope.launch { tooltipState.show() }
         },
     )
-
-    RecordButtonView(
-        isPressed = pressState.value is PressState.Pressing,
-        modifier = modifier
-            .pointerInput(Unit) {
-                awaitPointerEventScope {
-                    while (true) {
-                        val event = awaitPointerEvent()
-                        coroutineScope.launch {
-                            when (event.type) {
-                                PointerEventType.Press -> pressState.press()
-                                PointerEventType.Release -> pressState.release()
+    Box(modifier = modifier) {
+        RecordButtonView(
+            isPressed = pressState.value is PressState.Pressing,
+            modifier = Modifier
+                .pointerInput(Unit) {
+                    awaitPointerEventScope {
+                        while (true) {
+                            val event = awaitPointerEvent()
+                            coroutineScope.launch {
+                                when (event.type) {
+                                    PointerEventType.Press -> pressState.press()
+                                    PointerEventType.Release -> pressState.release()
+                                }
                             }
                         }
                     }
                 }
+        )
+        HoldToRecordTooltip(
+            modifier = Modifier.align(Alignment.CenterStart),
+            tooltipState = tooltipState,
+            anchor = {
+                // Don't anchor the tooltip box to the record button itself, as it will
+                // cause the tooltip to sit directly against the side of the screen.
```

Comment:
This was the simplest way I could think to implement the off-centre positioning. Right now `PopupPositionProvider` only supports customising the distance between the popup and the anchor.